controllers/erpControllers/invoiceController.js

```diff
@@ -155,6 +156,86 @@ methods.update = async (req, res) => {
   }
 };
 
+methods.summary = async (req, res) => {
+  try {
+    let defaultType = 'month';
+    
+    const { type } = req.query;
+
+    if (type) {
+      defaultType = type;
```

Comment:
I'm sorry for my unpolite comment, if not suitable for you, please just skip it.
What's happened if the ```defaultType``` is not week, month, or year? I think we need to catch and throw invalid params error here.

Reply:
No worries at all! You're right that handling cases where the defaultType parameter is not set to 'week', 'month', or 'year' is important. Thank you for pointing this out. I'll change the code.

Reply:
You are welcome